src/components/player/PlayerContainer.js

```diff
@@ -28,7 +29,11 @@ export class PlayerContainer extends Component {
     songDuration: 0,
     playingProgress: 0,
     volumeValue: 0.5,
-    isMuted: false
+    isMuted: false,
+    dialog: {
```

Comment:
please try to avoid nested objects in state where it is possible. In this case I do not see a reason why you need to save song to the state - you can get it from this.props at any moment.

Reply:
actually I saw another case at the bottom where you really need this `song`.  Ok. But then you do not need isOpen state. Having smth like `songToShare` is enough